src/api/posts/src/routes/posts.js

```diff
@@ -0,0 +1,111 @@
+  const defaultNumberOfPosts = process.env.MAX_POSTS_PER_PAGE || 30;
+  const capNumOfPosts = 100;
+  const page = parseInt(req.query.page || 1, 10);
+
+  let ids;
+  let perPage;
+  let postsCount;
+  let from;
+  let to;
+
+  /**
+   * Set 'perPage' to a value within the limits or
+   * to default if per_page is not present
+   */
+  if (req.query.per_page)
+    perPage = req.query.per_page > capNumOfPosts ? capNumOfPosts : req.query.per_page;
+  else perPage = defaultNumberOfPosts;
+
+  try {
+    postsCount = await getPostsCount();
+
+    /**
+     * Set the range of posts that will be requested
+     * {from, to}
+     */
+    from = perPage * (page - 1);
+    // Make sure the upper limit is not higher than the total number of posts in the DB
+    to = perPage * page > postsCount ? postsCount : perPage * page;
+
+    ids = await getPosts(from, to);
+  } catch (err) {
+    logger.error({ err }, 'Unable to get posts from Redis');
+    res.status(503).json({
+      message: 'Unable to connect to database',
+    });
+    return;
+  }
+
+  /**
+   * Add prev, next, first and last in the response's header.
+   * It's been implemented to work circularly.
+   * Once reached the last set of posts, 'next' points at the first set.
+   * Same case with 'prev' and the first set of posts.
+   */
+  const nextPage = to >= postsCount ? 1 : page + 1;
+  const prevPage = from === 0 ? Math.floor(postsCount / perPage) : page - 1;
+
+  res.set('X-Total-Count', postsCount);
+
+  res.links({
+    next: `/posts?per_page=${perPage}&page=${nextPage}`,
+    prev: `/posts?per_page=${perPage}&page=${prevPage}`,
+    first: `/posts?per_page=${perPage}&page=${1}`,
+    last: `/posts?per_page=${perPage}&page=${Math.floor(postsCount / perPage)}`,
+  });
+  res.json(
+    ids
+      // Return id and url for a specific post
+      .map((id) => ({
+        id,
+        url: `/posts/${id}`,
+      }))
+  );
+});
+
+posts.get('/:id', validatePostsIdParam(), async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const post = await Post.byId(id);
+
+    // If the object we get back is empty, use 404
+    if (!post) {
+      res.status(404).json({
+        message: `Post not found for id ${id}`,
+      });
+    } else {
+      switch (req.accepts(['json', 'text', 'html'])) {
+        case 'json':
+          res.append('Content-type', 'application/json').json(post);
+          break;
+        case 'text':
+          res.append('Content-type', 'text/plain').send(post.text);
+          break;
+        case 'html':
+          res.append('Content-type', 'text/html').send(post.html);
+          break;
+        default:
+          res.status(406).json({
+            message: 'Invalid content type',
+          });
+          break;
+      }
+    }
+  } catch (err) {
+    logger.error({ err }, 'Unable to get posts from Redis');
+    res.status(503).json({
+      message: 'Unable to connect to database',
+    });
+  }
+});
+
+module.exports = posts;
```

Comment:
Let's add a some more routes:

- `posts.post('/:id', ...)` for adding
- `posts.delete('/:id', ...)` for deleting.  Not sure if we need this one, or if we'd just delete at the feed level.  It's hard to remove a post, since it will just pop-up again the next time we crawl the feeds.
- `posts.put('/:id', ...)` for updating

All of these need to use the JWT protection I've added in Satellite (e.g., you have to be authenticated).  Furthermore, we'll need to deal with a user vs. admin:

- admin can do the operations above on any post (i.e., they don't need to own it)
- a user will only be able to do it for a post they own (i.e., belonging to a feed they own)

Reply:
Filed #1865 to handle these additional routes. This will likely involve talking with the User microservice or the Feed parsing microservice. Needs more discussion atm before we continue with this. Using the JWT protection in satellite should be easy enough